### PR TITLE
[enh] Allow to disable the backup during the upgrade

### PR DESCRIPTION
**Pull request by maniackcrudelis**
## The problem

Some people seem to have problems with the backup during the upgrade.

## Solution

Allow to not perform the backup during the upgrade process by setting the global variable $NO_BACKUP_UPGRADE.

## PR Status

Ready to review.
Tested with nextcloud.

## How to test

Set the variable $NO_BACKUP_UPGRADE during the upgrade.
```
sudo NO_BACKUP_UPGRADE=1 yunohost app upgrade nextcloud -f nextcloud_ynh/ --verbose
```

## Validation

- [ ] Principle agreement 1/2 : ljf
- [ ] Quick review 1/1 : ljf
- [ ] Simple test 0/1 : 
- [ ] Deep review 0/1 :

### CODE REVIEW

**Comment by maniackcrudelis** [comment]
Nothing more than that https://forum.yunohost.org/t/maj-nextcloud-11-vers-nextcloud-12-impossible/3421/10

**Comment by Psycojoker** [comment]
There is a conflict that needs to be resolved before we can merge this one :/

File: data/helpers.d/utils [hunk 1]
```diff
@@ -67,28 +74,34 @@ ynh_backup_before_upgrade () {
...
+				sudo yunohost backup delete $app_bck-pre-upgrade$old_backup_number > /dev/null
+			fi
+		else
+			ynh_die "Backup failed, the upgrade process was aborted."
 		fi
-	else
-		ynh_die "Backup failed, the upgrade process was aborted."
-	fi
+        else
+                echo "\$NO_BACKUP_UPGRADE is set, backup will be avoided. Be careful, this upgrade is going to be operated without securi$
```
Comment (julienmalik) [inline data/helpers.d/utils]: the end of this line seems buggy

Comment (maniackcrudelis) [reply data/helpers.d/utils]: Looks like a bad copy paste...
I'm fixing that

**Comment by Psycojoker** [comment]
@maniackcrudelis can you resolve the conflict so we can merge plz? I've tried to look at it but I'm not confident in doing it myself :/

**Comment by Psycojoker** [comment]
Thx :)

**Pull request merged**